google/cloud/storage/tests/curl_request_integration_test.cc

```diff
@@ -357,11 +357,13 @@ TEST(CurlRequestTest, UserAgent) {
   ASSERT_STATUS_OK(response);
   ASSERT_EQ(200, response->status_code) << "response=" << *response;
   auto payload = nlohmann::json::parse(response->payload);
-  ASSERT_EQ(1U, payload.count("headers"));
-  auto headers = payload["headers"];
-  EXPECT_THAT(headers.value("User-Agent", ""),
-              HasSubstr("test-user-agent-prefix"));
-  EXPECT_THAT(headers.value("User-Agent", ""), HasSubstr("gcloud-cpp/"));
+  ASSERT_TRUE(payload.is_object()) << "payload=" << response->payload;
+  ASSERT_TRUE(payload.contains("headers")) << "payload=" << response->payload;
+  auto const products = std::vector<std::string>(
+      absl::StrSplit(payload["headers"].value("User-Agent", ""), ' '));
+  EXPECT_THAT(products, AllOf(Contains("test-user-agent-prefix-1"),
```

Comment:
comment: I don't know what gives nicer errors between `AllOf(Contains(m1), Contains(m2), Contains(m3))` and `IsSupersetOf({m1, m2, m3})`

Reply:
I did not want to fiddle with the type of `{"foo", "bar", StartsWith("baz")}`, or `{Eq("foo"), Eq("bar"), StartsWith("baz")}`.  I think both are ambiguous (they need to be `initializer_list<T>` and `T` can be too many things.